validator/src/main/java/com/amazon/aoc/validators/LoadBalanceValidator.java

```diff
@@ -0,0 +1,170 @@
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.aoc.validators;
+
+import com.amazon.aoc.callers.ICaller;
+import com.amazon.aoc.enums.GenericConstants;
+import com.amazon.aoc.exception.BaseException;
+import com.amazon.aoc.exception.ExceptionCode;
+import com.amazon.aoc.fileconfigs.FileConfig;
+import com.amazon.aoc.helpers.MustacheHelper;
+import com.amazon.aoc.helpers.RetryHelper;
+import com.amazon.aoc.helpers.SortUtils;
+import com.amazon.aoc.models.Context;
+import com.amazon.aoc.models.SampleAppResponse;
+import com.amazon.aoc.models.ValidationConfig;
+import com.amazon.aoc.models.xray.Entity;
+import com.amazon.aoc.services.XRayService;
+import com.amazonaws.services.xray.model.Segment;
+import com.amazonaws.services.xray.model.Trace;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.PropertyNamingStrategy;
+import com.github.wnameless.json.flattener.JsonFlattener;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+public class LoadBalanceValidator implements IValidator {
+  private MustacheHelper mustacheHelper = new MustacheHelper();
+  private XRayService xrayService;
+  private ICaller caller;
+  private Context context;
+
+  private static final ObjectMapper MAPPER =
+      new ObjectMapper().setPropertyNamingStrategy(PropertyNamingStrategy.SNAKE_CASE);
+
+  // for unit test
+  public void setXRayService(XRayService xrayService) {
+    this.xrayService = xrayService;
+  }
+
+  @Override
+  public void init(
+      Context context, ValidationConfig validationConfig, ICaller caller, FileConfig expectedTrace)
+      throws Exception {
+    this.xrayService = new XRayService(context.getRegion());
+    this.caller = caller;
+    this.context = context;
+  }
+
+  @Override
+  public void validate() throws Exception {
+    // 2 retries for calling the sample app to handle the Lambda case,
+    // where first request might be a cold start and have an additional unexpected subsegment
+    int successes = 0;
+    while (successes < 5) {
+      // Call sample app and get locally stored trace
+      String traceId = this.getAppTraceId();
+      List<String> traceIdList = Collections.singletonList(traceId);
+      ;
+      log.info("value of Sample App traceId: {}", traceId);
+
+      // Retry 5 times to since segments might not be immediately available in X-Ray service
+      RetryHelper.retry(
+          5,
+          Integer.parseInt(GenericConstants.SLEEP_IN_MILLISECONDS.getVal()),
+          true,
+          () -> {
+            // get retrieved trace from x-ray service
+            Map<String, Object> retrievedTrace = this.getRetrievedTrace(traceIdList);
+            log.info("value of retrieved trace map: {}", retrievedTrace);
```

Comment:
Same comment here about logging to aggressively. I don't think this is needed.

Reply:
This comment could also helpful to see the entire trace returned to, for example, see why the test failed.

Reply:
Isn't the entire trace logged on failure?

Reply:
In this specific validator, the entire trace is not logged on failure.  That's why I believe it would be helpful to log the retrieved trace map every retry as the TraceValidator does.